airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -164,6 +165,9 @@ def get_dag_runs(
     else:
         query = query.filter(DagRun.dag_id == dag_id)
 
+    if state:
+        query = query.filter(DagRun.state.in_(state))
```

Comment:
```suggestion
    if states:
        query = query.filter(DagRun.state.in_(states))
```

Reply:
Hi @kaxil, thanks for your recommendations. I will change the datatype to Optional[List[str]], however I am not sure it is a good idea to change to plural as it is defined in singular in the specification v1.yml, even being an array:

    FilterState:
      in: query
      **name: state**
      schema:
        **type: array**
        items:
          type: string
      required: false
      description:
        The value can be repeated to retrieve multiple matching values (OR condition).

Reply:
After testing the change "Optional[List[str]]" I will need to change back the datatype to as with that change I can not pass the tests, errors detected in mypy checks:

airflow/plugins_manager.py:474: error: Incompatible types in assignment
(expression has type "Set[str]", variable has type "Optional[List[str]]")
            attrs_to_dump = PLUGINS_ATTRIBUTES_TO_DUMP
                            ^
airflow/plugins_manager.py:479: error: Item "None" of "Optional[List[str]]" has
no attribute "__iter__" (not iterable)

Reply:
Hi @kaxil, I just did the change on the datatype to Optional[List[str]] as you mentioned but now when I try to run he static ckecks I am getting hundreds of errors (mypy). Is this normal?, what am I doing wrong? Thanks so much for any help.

cc @ephraimbuddy, @uranusjr

Reply:
As long as CI passes it should be OK (not optimal, but OK if you don’t want to fix those).